auth.php

```diff
@@ -115,20 +115,31 @@ protected function _sql($id,$pw,$realm) {
 	*	@param $pw string
 	**/
 	protected function _ldap($id,$pw) {
-		$dc=@ldap_connect($this->args['dc']);
-		if ($dc &&
-			ldap_set_option($dc,LDAP_OPT_PROTOCOL_VERSION,3) &&
-			ldap_set_option($dc,LDAP_OPT_REFERRALS,0) &&
-			ldap_bind($dc,$this->args['rdn'],$this->args['pw']) &&
-			($result=ldap_search($dc,$this->args['base_dn'],
-				$this->args['uid'].'='.$id)) &&
-			ldap_count_entries($dc,$result) &&
-			($info=ldap_get_entries($dc,$result)) &&
-			@ldap_bind($dc,$info[0]['dn'],$pw) &&
-			@ldap_close($dc)) {
-			return $info[0][$this->args['uid']][0]==$id;
-		}
-		user_error(self::E_LDAP,E_USER_ERROR);
+	  $port=intval($this->args['port']?:389)
+	  $this->args['filter']=$this->args['filter']?:"uid=".$id;
+	  $filter=$this->args['filter']=$this->args['filter']?:"uid=".$id;
```

Comment:
2nd assignment repeated from one line above

Reply:
Damn - missed that. :(